sdk/instrumentation/google.golang.org/grpc/server.go

```diff
@@ -104,8 +104,9 @@ func wrapHandler(
 				if int(dataCaptureConfig.BodyMaxProcessingSizeBytes.Value) < len(reqBody) {
 					processingBody = reqBody[:dataCaptureConfig.BodyMaxProcessingSizeBytes.Value]
 				}
-				if filter.EvaluateBody(span, processingBody, md) {
-					return nil, status.Error(codes.PermissionDenied, "Permission Denied")
+				filterResult := filter.EvaluateBody(span, processingBody, md)
+				if filterResult.Block {
+					return nil, status.Error(codes.Code(filterResult.ResponseStatusCode), "Permission Denied")
```

Comment:
The status message will be different based on the status code. We probably need to map this correctly. We can have a static map for status code to status message based on https://developer.mozilla.org/en-US/docs/Web/HTTP/Status and just use that to get the correct message based on the status code.

Reply:
So what message should be I used incase the Status code is a non-standard one ?

Reply:
Wait. You mean if the customer uses a status code that's is not one of the HTTP standard ones eg. 472? You can just have something generic like "Request Error". But for the well known ones, we can defined a global map of status code to status message and use that.